sdk/iOS/OoyalaSkinSDK/React/OOReactSkinModel.m

```diff
@@ -268,7 +267,33 @@ - (void)handleDiscoveryClick:(NSString *)bucketInfo embedCode:(NSString *)embedC
                      bucketInfo:bucketInfo
                           pcode:self.player.pcode
                      parameters:nil];
-  [self.player setEmbedCode:embedCode];
+
+  __weak typeof(self) weakSelf = self;
+  void (^expectedBlock) (OOVideo *currentItem); //params of OOCurrentItemChangedCallback
+  expectedBlock = ^(OOVideo *currentItem) {
```

Comment:
I've noticed that this block is the same in `OOUpNextManager.m`. Should it be placed in a separate class? Or maybe it should go into `OoyalaSDK` since the logic does not rely on anything apart from `OyalaSDK`

Reply:
No, we can't move block into OOyalaSDK. Before I started changes related to this ticket I found that: there are a lot of cases in SkinSDK were embed code is set into player, but in some cases we can see pair of calls: `player setEmbedCode:`  and `player play` and in other cases just `player setEmbedCode:`. It means that not after every changing embed cod e we need to start playback automatically.